code/modules/reagents/reagent_dispenser.dm

```diff
@@ -390,4 +390,119 @@
 /obj/structure/reagent_dispensers/cauldron/can_transfer(var/obj/item/weapon/reagent_containers/R, var/mob/user)
 	if(user.a_intent != I_HELP)
 		return TRUE
-	return FALSE
\ No newline at end of file
+	return FALSE
+
+// BARRELS AND BARREL ACCESSORIES //
+/obj/structure/reagent_dispensers/cauldron/barrel
+	name = "metal barrel"
+	icon_state = "metalbarrel"
+	desc = "Originally used to store liquids & powder. It is now used as a source of comfort. This one is made of metal."
+	layer = TABLE_LAYER
+	flags = FPRINT | TWOHANDABLE | MUSTTWOHAND // If I end up being coherent enough to make it holdable in-hand
+	throwforce = 40 // Ends up dealing 20~ brute when thrown because thank you, based throw damage formula
+	var/list/exiting = list() // Manages people leaving the barrel
+
+/obj/structure/reagent_dispensers/cauldron/barrel/wood
+	name = "wooden barrel"
+	icon_state = "woodenbarrel"
+	desc = "Originally used to store liquids & powder. It is now used as a source of comfort. This one is made of wood."
+
+/obj/structure/reagent_dispensers/cauldron/barrel/update_icon()
+	return
+
+/obj/structure/reagent_dispensers/cauldron/barrel/can_pickup(mob/living/M)
+	if(!..())
+		return FALSE
+	if(M.get_strength() > 2)
+		return TRUE
+
+/obj/structure/reagent_dispensers/cauldron/barrel/kick_act(mob/living/carbon/human/H)
+	..()
+	if(reagents.total_volume > 10) //Beakersplashing only likes to do this sound when over 10 units
```

Comment:
if for some reason reagents is null this will runtime

Reply:
But why would reagents be null while the barrel is still able to be kicked?
Let this runtime if it happens.
weird review